middleware.go

```diff
@@ -252,11 +252,21 @@ func addCredentials(c *Client, r *Request) error {
 		}
 	}
 
-	// Token Auth
+	// Set the Authorization Header Scheme
+	var authScheme string
+	if !IsStringEmpty(r.AuthScheme) {
+		authScheme = r.AuthScheme
+	} else if !IsStringEmpty(c.AuthScheme) {
+		authScheme = c.AuthScheme
+	} else {
+		authScheme = "Bearer"
+	}
+
+	// Build the Token Auth header
 	if !IsStringEmpty(r.Token) { // takes precedence
-		r.RawRequest.Header.Set(hdrAuthorizationKey, "Bearer "+r.Token)
+		r.RawRequest.Header.Set(hdrAuthorizationKey, fmt.Sprintf("%s %s", authScheme, r.Token))
```

Comment:
I suspect that simple string concatenation will be more efficient.

Reply:
This [gist page](https://gist.github.com/dtjm/c6ebc86abe7515c988ec) has some data on this topic. It seems `Buffer`, `+`, and so on.  I will update it to `+`. Thanks.  

Done in my branch will push it to rigrassm:master